csharp/src/SeedLang/Common/TextRange.cs

```diff
@@ -50,5 +51,16 @@ public class TextRange : Range {
       }
       return Start == (range as TextRange).Start && End == (range as TextRange).End;
     }
+
+    public static TextRange RangeOfToken(IToken token) {
+      return RangeOfTokens(token, token);
+    }
+
+    public static TextRange RangeOfTokens(IToken start, IToken end) {
+      // TODO: need scan the source string to calculate the end column if the end token is in
+      // multiple lines.
+      return new TextRange(start.Line, start.Column,
```

Comment:
I feel it's not good that the TextRange class in Common component depends on Antlr4.Runtime. Maybe we can limit this kind of dependency in X and Block components. It's better to keep the Token to TextRange conversion in SeeLang.X or SeedLang.Block.

Reply:
Done. Moved the two methods to a utility class in SeedLang.X.